### ISSUE TITLE
Support temporary stack policies in AWS

### ISSUE THREAD

**Issue by kmjennison** [serverless/serverless#6182]
# This is a Feature Proposal

## Description

It would be nice to be able to use [temporary stack policies](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/protect-stack-resources.html#protect-stack-resources-updating) to allow one-off modification of resources that are typically protected.

This is helpful when 1) the stack policy is intentionally restrictive to prevent any accidental resource modification, and 2) you want to modify one of those resources.

Currently with Serverless, I believe the only way to handle this is to update the stack policy, deploy the resource modification, and then update the stack policy again. This is cumbersome and also causes risks if you forget to revert the stack policy or if someone unintentionally modifies a resource in the timeframe when the policy is less restrictive.

Serverless already supports stack policies (#1952).

Ideally, we would be to be able to specify a `temporaryStackPolicy` in serverless.yml that runs for only the current update, using [--stack-policy-during-update-body](https://docs.aws.amazon.com/cli/latest/reference/cloudformation/update-stack.html). It would have to rely on some deployment state (an ID, timestamp, hash?) to only use the temporary stack policy temporarily.

I don't follow the Serverless Framework closely enough to have a good suggestion on how to implement this, but I'd love to hear whether others would find the feature helpful or other ways of approaching the problem.

**Comment by nadobando** [comment]
+1

**Comment by neverendingqs** [comment]
It looks like a bulk of the changes will need to go into https://github.com/serverless/serverless/blob/master/lib/plugins/aws/lib/updateStack.js

**Comment by claydanford** [comment]
I've run into a need for this today, perhaps others can assist with a work around.

We protect dynamodb tables through stack policy. Now we have a valid use case for when we actually want to delete a table but no way to override using CI. In fact the best we can do I think is to modify the policy, update the stack, change it back. Very messy.

**Comment by ankon** [comment]
We're also using stack policies to protect critical resources (DynamoDB tables, bucket, and user pools).

> Ideally, we would be to be able to specify a temporaryStackPolicy in serverless.yml that runs for only the current update, using --stack-policy-during-update-body.

Putting it into `serverless.yml` feels odd: If this is temporary, for the time of deploying, I would expect this to be a parameter given on the command line: `serverless deploy --update-policy /path/to/policy.json` (or provide it directly as a body).

**Comment by kmjennison** [comment]
@ankon I agree. Since I opened this issue, we've been handling this manually by updating the stack policy via API before a deploy, which feels more in line with the goal of making only intentional changes. The stack policy then reverts during deploy to the one in `serverless.yml`. One downside is that this requires knowing when the stack-modifying deploy will run (it works for our small team but might not for a larger one). Personally, I'd also prefer those changes to live in code history.
